subsys/bluetooth/host/l2cap.c

```diff
@@ -2972,6 +2975,41 @@ int bt_l2cap_ecred_chan_reconfigure(struct bt_l2cap_chan **chans, uint16_t mtu)
 		if (mtu < BT_L2CAP_LE_CHAN(chans[i])->rx.mtu) {
 			return -EINVAL;
 		}
+
+#if defined(CONFIG_BT_L2CAP_SEG_RECV)
+		/**
+		 * If more than one channel is being configured, the MPS field
+		 * shall be greater than or equal to the current MPS size of
+		 * each of these channels. If only one channel is being
+		 * configured, the MPS field may be less than the current MPS
+		 * of that channel.
+		 */
+		if (BT_L2CAP_LE_CHAN(chans[i])->chan.ops->seg_recv) {
+			seg_direct = true;
+		}
+#endif
+
+		if (i == 0) {
+			mps = BT_L2CAP_LE_CHAN(chans[i])->rx.mps;
+		}
+
+		if (i > 0) {
+			/* For channels using segment direct API we can only
+			 * decrease MPS and this is not allowed for multiple
+			 * channels. So if any channel is using this MPS cannot
+			 * change.
```

Comment:
But aren't we here only checking if any previous handled channels use `seg_direct`? 

For example if `chans[1]` does not use it, then it is false here and the `if (mps != BT_L2CAP_LE_CHAN(chans[i])->rx.mps) {` isn't applied, even if `chans[2]` uses it. 

For this check to make sense, don't we need to iterate on all streams first to set the `seg_direct` before we start doing this check?

Reply:
for channels not using seg direct API it is possible to increase MPS (ie if previously it was clipped due to small MTU),  so this check matters only if we step on channel using seg direct

I'll see if those can be more explicit, but initially wanted to keep changes minimal and would multiple loops (but maybe this doesn't hurt)